Plugin.cs

```diff
@@ -1,39 +1,64 @@
+    public const string modName = "LC Always Hear Active Walkies";
+    public const string modVersion = "1.4.2";
+}
+
+[BepInPlugin(PluginInfo.modGUID, PluginInfo.modName, PluginInfo.modVersion)]
+public class Plugin : BaseUnityPlugin
+{
+    private ConfigEntry<float> configAudibleDistance;
+    public static float AudibleDistance;
+        
+    private ConfigEntry<float> configWalkieRecordingRange;
+    public static float WalkieRecordingRange;
+
+    private ConfigEntry<float> configPlayerToPlayerSpatialHearingRange;
+    public static float PlayerToPlayerSpatialHearingRange;
+
+    public static ManualLogSource Log;
+
+    private static Plugin Instance;
+
+    public void Awake()
     {
-        public static ManualLogSource Log;
-        private const string modGUID = "suskitech.LCAlwaysHearActiveWalkie";
-        private const string modName = "LC Always Hear Active Walkies";
-        private const string modVersion = "1.4.2";
-        private readonly Harmony harmony = new Harmony(modGUID);
-        private static LCAlwaysHearWalkieMod Instance;
-
-        private void Awake()
-        {
-            if (Instance == null) {
-                Instance = this;
-            }
-            Log = BepInEx.Logging.Logger.CreateLogSource(modGUID);
-            
-            Log.LogInfo("\\ /");
-            Log.LogInfo("/|\\");
-            Log.LogInfo(" |----|");
-            Log.LogInfo(" |[__]| Always Hear Active Walkies");
-           Log.LogInfo($" |.  .| Version {modVersion} Loaded");
-            Log.LogInfo(" |____|");
-
-            harmony.PatchAll(typeof(LCAlwaysHearWalkieMod));
-            harmony.PatchAll(typeof(PlayerControllerBPatch));
-            harmony.PatchAll(typeof(WalkieTalkiePatch));
+        if (Instance == null) {
+            Instance = this;
         }
+
+        configAudibleDistance = Config.Bind("General", "AudibleDistance", 20f, "");
+        AudibleDistance = configAudibleDistance.Value;
+            
+        configWalkieRecordingRange = Config.Bind("General", "WalkieRecordingRange", 20f, "");
+        WalkieRecordingRange = configWalkieRecordingRange.Value;
+            
+        configPlayerToPlayerSpatialHearingRange = Config.Bind("General", "PlayerToPlayerSpatialHearingRange", 20f, "");
+        PlayerToPlayerSpatialHearingRange = configPlayerToPlayerSpatialHearingRange.Value;
+
+        Log = BepInEx.Logging.Logger.CreateLogSource(PluginInfo.modGUID);
+
+        Harmony harmony = new(PluginInfo.modGUID);
+
+        harmony.PatchAll(typeof(Plugin));
+        harmony.PatchAll(typeof(PlayerControllerBPatch));
+        harmony.PatchAll(typeof(WalkieTalkiePatch));
+        harmony.PatchAll(typeof(StartOfRoundPatch));
+
+
+        Log.LogInfo("\\ /");
+        Log.LogInfo("/|\\");
+        Log.LogInfo(" |----|");
+        Log.LogInfo(" |[__]| Always Hear Active Walkies");
+        Log.LogInfo($" |.  .| Version {PluginInfo.modVersion} Loaded");
+        Log.LogInfo(" |____|");
     }
-}
+}
```

Comment:
I'm loving all of these configuration changes and thanks for catching my namespace mistake. Would it be possible to break _only_ the configuration and piece into a separate PR so we can test and integrate configurable distances separate from the spatial audio changes?

Reply:
I ended up with some time tonight so resolved the conflicts and integrated these config changes. I've attributed you in the README, thanks very much!